isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -2178,98 +2403,251 @@ namespace isobus
 
 				case static_cast<std::uint32_t>(CANLibParameterGroupNumber::VirtualTerminalToECU):
 				{
-					switch (message->get_data().at(0))
+					switch (data.at(0))
 					{
 						case static_cast<std::uint8_t>(Function::SoftKeyActivationMessage):
 						{
-							std::uint8_t keyCode = message->get_data().at(1);
+							std::uint8_t keyCode = data.at(1);
 							if (keyCode <= static_cast<std::uint8_t>(KeyActivationCode::ButtonPressAborted))
 							{
+								if (parentVT->get_vt_version_supported(VTVersion::Version6))
+								{
+									//! @todo process TAN
+								}
+
 								parentVT->process_softkey_event_callback(static_cast<KeyActivationCode>(keyCode),
-								                                         static_cast<std::uint16_t>(message->get_data().at(6)),
-								                                         (static_cast<std::uint16_t>(message->get_data().at(2)) | static_cast<std::uint16_t>(message->get_data().at(3) << 8)),
-								                                         (static_cast<std::uint16_t>(message->get_data().at(4)) | static_cast<std::uint16_t>(message->get_data().at(5) << 8)),
+								                                         static_cast<std::uint16_t>(data.at(6)),
+								                                         (static_cast<std::uint16_t>(data.at(2)) | static_cast<std::uint16_t>(data.at(3) << 8)),
```

Comment:
One last note, when we cast to `std::uint16_t` like we do here, the cast should happen before the bit shift to avoid shifting all the bits out of the `std::uin8t_t`. So, it's more correct to do `(static_cast<std::uint16_t>(data.at(3)) << 8)` instead of  ` static_cast<std::uint16_t>(data.at(3) << 8)` Check out the case for `Function::VTStatusMessage` for an example